Sources/HAP/Server/Device.swift

```diff
@@ -77,14 +80,14 @@ public class Device {
     ///   - accessories: accessories to be bridged
     convenience public init(
         bridgeInfo: Service.Info,
-        setupCode: String,
         storage: Storage,
-        accessories: [Accessory]) {
+        accessories: [Accessory],
+        defaultSetupCode: String = "automatic") {
```

Comment:
I don't like having magic strings; we could have an enum with an associated value.

```
enum SetupCode {
    case automatic
    case predefined(String)
}

init(setupCode: SetupCode = .automatic)
```